src/org/zaproxy/zap/extension/custompages/CustomPage.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.custompages;
+
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.utils.EnableableInterface;
+
+public interface CustomPage extends EnableableInterface{

```

Comment:
Maybe change the interface to just have the method `isCustomPage`? (And let `EnableableInterface` and the other methods be part of the default implementation?)
This would allow to have simpler/immutable default implementations (for 200, 404, 500).

Reply:
Per comments in IRC, not worth changing.